Source/Immutable/Private/Immutable/ImmutablePassport.cpp

```diff
@@ -568,48 +545,46 @@ void UImmutablePassport::OnConnectPKCEResponse(FImtblJSResponse Response) {
 }
 #endif
 
-void UImmutablePassport::OnGetIdTokenResponse(FImtblJSResponse Response) {
-  if (auto ResponseDelegate = GetResponseDelegate(Response)) {
-    // Extract the credentials
-    auto Credentials =
-        JsonObjectToUStruct<FImmutablePassportTokenData>(Response.JsonObject);
+void UImmutablePassport::OnGetIdTokenResponse(FImtblJSResponse Response)
+{
+  if (auto ResponseDelegate = GetResponseDelegate(Response))
+  {
+    const FString IdToken = Response.JsonObject->GetStringField(TEXT("result"));
+        
+    if (!Response.success || IdToken.IsEmpty())
+    {
+      IMTBL_LOG("No stored Id token found.");
 
-    if (!Response.success || !Credentials.IsSet() ||
-        !Credentials->idToken.Len()) {
-      IMTBL_LOG("No stored credentials found.");
-      FString Msg;
-      Response.Error.IsSet()
-          ? Msg = Response.Error->ToString()
-          : Msg = Response.JsonObject->GetStringField(TEXT("error"));
-      ResponseDelegate->ExecuteIfBound(
-          FImmutablePassportResult{false, Msg, Response});
-    } else {
-      IMTBL_LOG("Stored ID token found.");
-      ResponseDelegate->ExecuteIfBound(
-          FImmutablePassportResult{true, Credentials->idToken});
+      const FString Msg = Response.Error.IsSet() ? Response.Error->ToString() : Response.JsonObject->GetStringField(TEXT("error"));
+
+      ResponseDelegate->ExecuteIfBound(FImmutablePassportResult{false, Msg, Response});
+    }
+    else
+    {
+      IMTBL_LOG("Stored Id token found.");
+      ResponseDelegate->ExecuteIfBound(FImmutablePassportResult{true, IdToken});
     }
   }
 }
 
-void UImmutablePassport::OnGetAccessTokenResponse(FImtblJSResponse Response) {
-  if (auto ResponseDelegate = GetResponseDelegate(Response)) {
-    // Extract the credentials
-    auto Credentials =
-        JsonObjectToUStruct<FImmutablePassportTokenData>(Response.JsonObject);
+void UImmutablePassport::OnGetAccessTokenResponse(FImtblJSResponse Response)
+{
+  if (auto ResponseDelegate = GetResponseDelegate(Response))
+  {
+    const FString AccessToken = Response.JsonObject->GetStringField(TEXT("result"));
 
-    if (!Response.success || !Credentials.IsSet() ||
-        !Credentials->accessToken.Len()) {
-      IMTBL_LOG("No stored credentials found.");
-      FString Msg;
-      Response.Error.IsSet()
-          ? Msg = Response.Error->ToString()
-          : Msg = Response.JsonObject->GetStringField(TEXT("error"));
-      ResponseDelegate->ExecuteIfBound(
-          FImmutablePassportResult{false, Msg, Response});
-    } else {
+    if (!Response.success || AccessToken.IsEmpty())
```

Comment:
From testing, `success` can be true and `AccessToken` can be null. It's currently throwing an error when this happens. Please add this check to both Get Access Token and Get ID Token.

Reply:
Ok, I have removed the empty token check and edited the log messages

Reply:
It still doesn't work. Also, I don't think you should remove that empty token check because it can still be empty.

It looks like the error is before `OnGetAccessTokenResponse` is called. It's complaining that `result` is not in the JSON object.